command/agent/fs_endpoint.go

```diff
@@ -379,16 +379,23 @@ func (s *StreamFramer) ExitCh() <-chan struct{} {
 	return s.exitCh
 }
 
+// Err returns the error that caused the StreamFramer to exit
+func (s *StreamFramer) Err() error {
+	s.l.Lock()
+	defer s.l.Unlock()
+	return s.err
+}
+
 // run is the internal run method. It exits if Destroy is called or an error
 // occurs, in which case the exit channel is closed.
 func (s *StreamFramer) run() {
 	var err error
 	defer func() {
-		close(s.exitCh)
 		s.l.Lock()
 		s.running = false
-		s.Err = err
+		s.err = err
```

Comment:
Should this be set using a `SetErr()` method that has guards against concurrent access?

Reply:
The appropriate lock is acquired on L394 and this should be the only place that sets `s.err` so I don't want to add a helper to encourage other uses.